### ISSUE TITLE
Custom type guard in getters is broken

### ISSUE THREAD

**Issue by tinganho** [Microsoft/TypeScript#7150]
**TypeScript Version:** 

nightly

**Code**

```ts
class A {
	get a(): this is A {
		return true
	}
}
```

**Expected behavior:**

The getter type guard should not complain

**Actual behavior:** 
```
tsc
Library/DataStore.ts(55,27): error TS1228: A type predicate is only allowed in return type position for functions and methods.
Library/DataStore.ts(103,20): error TS1228: A type predicate is only allowed in return type position for functions and methods.
```

**Comment by RyanCavanaugh** [comment]
Is this a regression?

**Comment by mhegazy** [comment]
this is by design. typeguards only exist on signatures, and not properties. getters/setters are properties, and that caused different issues. this is a recent change, so if used nightlies before, then it would have worked. see https://github.com/Microsoft/TypeScript/pull/7067 for more details.

**Comment by saneyuki** [comment]
> this is by design. typeguards only exist on signatures, and not properties. getters/setters are properties, and that caused different issues. 

I seem property cannot be a typeguard would be a problem in some cases of an union type (e.g. `Option<T>` type). They make the property which checks the current type a constant value instead of calling a method to optimize the performance.

code example:
```typescript
type Option<T> = Some<T> | None<T>;

class Some<T> {
  constructor(private v: T) {
    this.isSome = true;
  }

  isSome: this is Some<T>;
}

class None<T> {
  constructor() {
    this.isSome = false;
  }

  isSome: this is Some<T>;
}

// ---------------------------

if (option.isSome) {
  // some
}
else {
  // none
}
```
